### ISSUE TITLE
NLP2 - Assignment Project 2 Todo's

### ISSUE THREAD

**Issue by JMitnik** [JMitnik/NLP-Posterior-Collapse-and-Probing#6]
# Code Iteration 1
---
### Control tasks
- [x] Setup a control task for the Simple Probe [**Pieter**]
- [x] Tree-based control task: [**Jonathan**]
  - [x] Think of a way to do control tasks for the Structural Probe...
  - [x] ... or we Dependency graph parsing as is done in the paper

### Simple Probes: Fine-Tuning
- [x] Add validation scores to Skorch methods [**Pieter**]

### Structural probe: Fine-tuning
- [ ] Add in some hyperparameter tuning [**Jonathan**]

### Results
- [x] **Make CSVs** entire pipeline work to _compare_ LSTM and Transformer (store results in csv with Config) [**Pieter**]
	- [ ] UUID / Timestamps for when we run this
	- [x] Have csv's for:
		- [x] Training / Validation POS probe
		- [ ] Training structural probe
		- [x] Validation structural probe
		- [ ] Test structural probe
    - [x] We expect different models
	- [x] ❓  Figure out whether we want to add control tasks in separate CSVs.
- [x] Plot fancy trees for dependencies (Use the true values from Conll, and somehow parse the predicted into similar format) [**Pieter/Jonathan**]
 (See Comments)

# Poster (Start: Tuesday)
---
### Conclusion / Discussion

# Code Iteration 2
---
### Debug results
- [ ] Debug the distances: check that the distances we see in the corpus correspond correctly to their output from `create_gold_distances`.
- [ ] Debug the UUAS score as well; it doesn't seem like it changes as much as how much the loss is changing over the training\
- [ ] Why eval not overfit?

### Bonus cool part (after poster, presumably)
- [ ] Add some interesting component (instead of part-of-speech, we try to do something like **coreference**).
- [ ] Use a similar control task as above for this.
